### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by Huizhen-Tan**
Most of the tasks are completed.
Try first pull request and try to get some feedbacks.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Code appear valid for Easy Difficulty 
- Results appear valid and good enough
- Sufficiently detailed commit messages

Other Feedback
- clean up unused import (e.g. `from tensorflow.keras.layers import Conv2D, Activation, BatchNormalization, \
    Dropout, Input, concatenate, Add, UpSampling2D, Conv2DTranspose` from the main)
- clean up code that has been commented out and would not be used again
- good code structure and modularisation

**Comment by rexsw** [comment]
Code appear valid for Normal Difficulty

Some good comments, but long functions like def improved_unet(h, w) could use comments breaking down the model to make it clear which parts of the unet are being built when

Rather then using pythons random module and numpy, tensorflow has dataloaders that you can use

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme good
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. Improve Pull Request description.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.
